api/v1alpha1/tenantcontrolplane_funcs.go

```diff
@@ -65,6 +65,16 @@ func (in *TenantControlPlane) DeclaredControlPlaneAddress(ctx context.Context, c
 			if ip := lb.IP; len(ip) > 0 {
 				return ip, nil
 			}
+			if hostname := lb.Hostname; len(hostname) > 0 {
+				// Resolve hostname to IP address
+				ips, err := net.LookupIP(hostname)
+				if err != nil {
+					return "", errors.Wrap(err, "cannot resolve LoadBalancer hostname to IP")
+				}
+				if len(ips) > 0 {
+					return ips[0].String(), nil
+				}
+			}
```

Comment:
Kubernetes and FQDNs have a tough relationship, unfortunately.

The problem here is that we don't know which resolver is responsible for the DNS resolution and it could differ from environment to environment.

The Tenant Control Plane address extracted here is going to be used as the `kubernetes.default.svc` endpoint address, which is then resolved by kubelet to forward traffic to the API Server from internal pods. The problem with this approach resides in the fact the DNS resolution could be changed over time, bringing the cluster Pod to the Kubernetes API Server connection broken.

At the current state, our suggestion is setting a Virtual IP for the given hostname by manually defining it, since the implications behind L7 routing with the API Server must be well known by the users: although the aim of this PR is to simplify it, we could end-up with users complaining _it doesn't work_ despite it's more one of their problems in not understanding the complexities of the topic.

Furthermore, an Ingress exposition requires a TLS passthrough for the client-based certificates authentication, again something that could be hard to understand for newbies, without considering that routing would be based on hostname: if an Ingress Controller is listening on `1.2.3.4` and used by several TCPs (`tcp-alpha.my.corp.tld`, `tcp-bravo.my.corp.tld`, `tcp-charlie.my.corp.tld`) the routing would be broken, unless using a routing based on the port number, making totally useless the Ingress.

Reply:
Thank you for your detailed explanation about the challenges of using FQDNs with Kubernetes in the context of Tenant Control Plane addressing. I understand the complexities and potential issues arising from DNS resolution inconsistencies.

To address these concerns, I've implemented the following:

- A new getLoadBalancerAddress function to handle IP address extraction and hostname checks for LoadBalancer ingress.
- Comprehensive comments explaining why hostnames aren't supported, potential risks, and recommended solutions.
- A concise error message: "hostname not supported for LoadBalancer ingress: use static IP instead".

These changes aim to prevent misconfigurations and help users quickly identify and resolve issues related to hostname usage in this context. The detailed comments serve as documentation for developers, while the error message provides immediate, actionable feedback to users.

Let me know if you'd like any further modifications or have any questions about this implementation.